File: requirement/prototype/frontend/src/types/process.ts
import { BaseEntity } from './common';

/**
 * 工序（Process）类型定义
 * 用于售前调研阶段，按项目维护工序信息
 */
export interface Process extends BaseEntity {
  name: string;                      // 工序名称
  projectId: string;                 // 所属项目ID
  projectName?: string;              // 所属项目名称（冗余字段，便于显示）
  code?: string;                     // 工序编号（可选）
  description?: string;              // 工序描述
  sortOrder?: number;                // 排序序号
}

File: source/backend/src/main/java/com/dataacquisition/modules/process/entity/Process.java
package com.dataacquisition.modules.process.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.dataacquisition.common.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 工序实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_process")
public class Process extends BaseEntity {

    /**
     * 主键
     */
    @TableId(type = IdType.AUTO)
    private Long id;

    /**
     * 工序名称
     */
    private String name;

    /**
     * 所属项目ID
     */
    @TableField("project_id")
    private Long projectId;

    /**
     * 所属项目名称（非数据库字段，用于展示）
     */
    @TableField(exist = false)
    private String projectName;

    /**
     * 所属车间ID
     */
    @TableField("workshop_id")
    private Long workshopId;

    /**
     * 工序编号
     */
    private String code;

    /**
     * 工序描述
     */
    private String description;

    /**
     * 排序序号
     */
    @TableField("sequence_no")
    private Integer sortOrder;
}
